ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Cleaner.java

```diff
@@ -321,6 +321,9 @@ private boolean removeFiles(String location, ValidWriteIdList writeIdList, Compa
       ci.setWriteIds(dir.getAbortedWriteIds());
     }
     obsoleteDirs.addAll(dir.getAbortedDirectories());
+    if (isDynPartAbort(table, ci)) {
+      obsoleteDirs = dir.getAbortedDirectories();
```

Comment:
Could we add a small explanation comment here why obsoleteDirs has to be overwritten for isDynPartAbort case?